staging/src/k8s.io/apiserver/pkg/endpoints/watch_test.go

```diff
@@ -591,6 +591,7 @@ func TestWatchHTTPErrors(t *testing.T) {
 	}
 
 	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		defer watcher.Stop()
```

Comment:
Why is this call needed ?

Reply:
`watcher.Stop` has moved to `serveWatch` function. The test codes now is tricky, but I have not found a better way to call `watcher.Stop` in test codes.

Reply:
But you're stopping the watcher on _any_ request, instead of just the relevant ones.